app/src/main/java/com/ikarabulut/Rules.java

```diff
@@ -15,15 +15,15 @@ public static boolean willBeAliveNextGeneration(int aliveNeighborsCount, boolean
         return isAliveNextGeneration;
     }
 
-    private static void isUnderpopulated() {
+    public void isUnderpopulated() {
```

Comment:
Do these methods need to be public? I think they're only being used by the `willBeAliveNextGeneration` method and no outside methods.

Reply:
I can set an interface method to private but then I would need to have a body for the method in the interface. 

The issue is that I cannot set an interface variable to something else because if I add a variable to an interface then it is `final`. 

This did actually open up my mind to a simpler implementation to these rules which Ill be pushing up shortly.

Basically I set each interface method to `private` and had each return a `boolean`. Then in the `willBeAliveNextGeneration()` method I can just return the `isUnderpoulated() || isFertileLand() || isOverpopulated()`

Reply:
Oh interesting. I left some comments with a discussion about trade-offs.  But TIL on interface implementations. Thanks!

(I'm not asking for any changes with my comments about inheritance. They are thought experiments and something to try and see what happens when you have time.)